classes/Rest/XdmodApplicationFactory.php

```diff
@@ -170,12 +190,37 @@ public static function getInstance()
 
         // SETUP: error handler
         $app->error(function (\Exception $e, $code) use ($app) {
-                $exceptionOutput = \handle_uncaught_exception($e);
-                return new Response(
-                    $exceptionOutput['content'],
-                    $exceptionOutput['httpCode'],
-                    $exceptionOutput['headers']
-                );
+            if($code == 405 && strtoupper($_SERVER['REQUEST_METHOD']) === 'OPTIONS'){
+                try {
+                    $corsDomains = \xd_utilities\getConfiguration('cors', 'domains');
+                } catch (\Exception $cors) {
+                    $corsDomains = null;
+                }
+                if (!empty($corsDomains)){
+                    $allowedCorsDomains = explode(',', $corsDomains);
+                    $origin = $_SERVER['HTTP_ORIGIN'];
+                    if (in_array($origin, $allowedCorsDomains)) {
+                        // if these headers change we will need to update the `after` above
+                        return new Response(
+                            '',
+                            204, /* ignored use header `X-Status-Code` */
+                            [
+                                'X-Status-Code' => 204,
+                                'Vary' => 'Origin',
+                                'Access-Control-Allow-Origin' => $origin,
+                                'Access-Control-Allow-Headers' => 'x-requested-with',
+                                'Access-Control-Allow--Credentials' => 'true'
```

Comment:
Is this a typo? (Different from the one above)
```suggestion
                                'Access-Control-Allow-Credentials' => 'true'
```